### PR TITLE
[XPU] Updates for XPU Graph + torch.compile compatibility

### PR DESCRIPTION
**Pull request by CaoE**
## Motivation

This change updates for XPU Graph + torch.compile support compatibility.

## Changes
* `mha_fwd` optional out argument — callers can now pass a pre-allocated output tensor. Under graph capture, SGLang binds a stable output buffer once and reuses it across replays.
* `fwd` return type and schema — the top-level op now returns a fixed 4-tuple `(out, softmax_lse, out_accum, softmax_lse_accum)` instead of `Tensor[]`. The Torch schema is annotated as `Tensor(a!)? out=None) -> (Tensor(a!), Tensor, Tensor, Tensor)`, so functionalization correctly models the in-place output aliasing required for Dynamo tracing.
* `weak_ref_tensor` op for XPU — registers a no-ownership view over an XPU tensor (`at::from_blob`). SGLang uses it to release captured-output ownership earlier on the XPU graph path.
* Minor change— `cu_seqlens_q == None` → `cu_seqlens_q is None` for torch.compile compatibility.

### CODE REVIEW

File: tests/test_weak_ref_tensor.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
+import pytest
+import torch
+import utils
+
+device = utils.get_device()
```
[the remark this answers is not in the document]
Comment (CaoE) [reply tests/test_weak_ref_tensor.py:5]: This follows other files.

File: python/sgl_kernel/flash_attn.py [hunk 2]
```diff
@@ -245,7 +255,7 @@ def flash_attn_with_kvcache(
     rotary_cos, rotary_sin = [maybe_contiguous(x) for x in (rotary_cos, rotary_sin)]
     rotary_seqlens = maybe_contiguous(rotary_seqlens)
 
-    if cu_seqlens_q == None:  # !is_varlen_q
+    if cu_seqlens_q is None:  # !is_varlen_q
         cu_seqlens_q = torch.arange(
             0, q.size(0) + 1, dtype=torch.int, device=q.device
         ) * q.size(1)
```
[the remark this answers is not in the document]
Comment (CaoE) [reply python/sgl_kernel/flash_attn.py:261]: I just changed `cu_seqlens_q i== None`  to `cu_seqlens_q is None` , and wouldn't touch other parts.

File: src/sycl/flash_attention.cpp [hunk 3]
```diff
@@ -1240,12 +1248,24 @@ std::vector<at::Tensor> mha_fwd(
...
-    int const sm_margin) {
+    int const sm_margin,
+    std::optional<at::Tensor>& out_) {
   TORCH_CHECK(cu_seqlens_k.data_ptr<int>() != nullptr, "cu_seqlens_k is not valid.");
+  if (out_.has_value()) {
+    const at::Tensor& out_val = out_.value();
+    TORCH_CHECK(out_val.scalar_type() == q.scalar_type(), "out dtype must match q dtype");
+    TORCH_CHECK(
+        out_val.dim() == 3 && out_val.size(0) == q.size(0) && out_val.size(1) == q.size(-2) &&
+            out_val.size(2) == v.size(-1),
```
Comment (mingfeima) [inline src/sycl/flash_attention.cpp:1259]:
```suggestion
    TORCH_CHECK(
        out_val.dim() == 3 && out_val.size(0) == q.size(0) && out_val.size(1) == q.size(1) &&
            out_val.size(2) == v.size(2),
```
more readable, less brain work.

File: src/weak_ref_tensor.cpp [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#include <ATen/ATen.h>
+#include <ATen/Tensor.h>
+
+#include <vector>
+
+at::Tensor weak_ref_tensor(const at::Tensor& tensor) {
```
Comment (mingfeima) [inline src/weak_ref_tensor.cpp:24]: better give a [NOTE] why need this.

**Review by mingfeima — CHANGES_REQUESTED** [review]
generally LGTM, just some minor issues to address.

File: src/sycl/flash_attention.cpp (hunk 3 above)
Comment (CaoE) [reply src/sycl/flash_attention.cpp:1259]: Thanks for your commemts. Revised.

File: src/weak_ref_tensor.cpp (hunk 4 above)
Comment (CaoE) [reply src/weak_ref_tensor.cpp:24]: Added.

File: src/sycl/flash_attention.cpp (hunk 3 above)
Comment (CaoE) [reply src/sycl/flash_attention.cpp:1259]: I just realized that q might be 3/4D. Changed it back.

Comment (mingfeima) [reply src/sycl/flash_attention.cpp:1259]: here it checks `out_val.dim() == 3`, does out needs to be the same dimension as q?

**Comment by CaoE** [comment]
@copilot resolve the merge conflicts in this pull request

File: src/sycl/flash_attention.cpp (hunk 3 above)
Comment (CaoE) [reply src/sycl/flash_attention.cpp:1259]: The agent reviewed the entire C++ implementation, stating that q is fixed at 3D, while v could be 3D or 4D. The comments have been updated.

**Review by mingfeima — APPROVED** [review]
